.changeset/quick-onions-leave.md

```diff
@@ -0,0 +1,34 @@
+---
+'astro': patch
+---
+
+Implements legacy content and data collections using `glob()` loader
+
+:warning: **BREAKING CHANGE FOR LEGACY CONTENT COLLECTIONS** :warning:
+
+By default, collections that use the the old types (content or data) are now implemented using the glob loader, with extra backward-compat handling. This includes any collection without a `loader` defined.
+
+Any legacy content collections are handled like this:
+
+- a `glob` loader collection is defined, with patterns that match the previous handling (matches `src/content/<collection name>/**/*.md` and other content extensions depending on installed integrations, with underscore-prefixed files and folders ignored)
+- When used in the runtime, the entries have an ID based on the filename in the same format as legacy collections
+- A `slug` field is added with the same format as before
+- A `render()` method is added to the entry, so they can be called using `entry.render()`
+- `getEntryBySlug` is supported
+
+Legacy data collections are handled like this:
+
+- a `glob` loader collection is defined, with patterns that match the previous handling (matches `src/content/<collection name>/**/*{.json,.yaml}` and other data extensions, with underscore-prefixed files and folders ignored)
+- Entries have an ID that is not slugified
+- `getDataEntryById` is supported
```

Comment:
I don't think this one and `getEntryBySlug` should be supported anymore. They're deprecated already IIRC so we should just remove them

Reply:
I only implemented them because this is meant for backwards-compatibility. Once they move to content layer-proper, they're not supported anymore.